### PR TITLE
Tracking progress in the skill map

### PR DESCRIPTION
**Pull request by riknoll**
Fixes a few bugs around saving projects and also actually tracks the activity progress/completion. Still need to hook up the completion modal and add the debug hooks for marking activities as completed

### CODE REVIEW

File: skillmap/src/components/SkillCard.tsx [hunk 1]
```diff
@@ -62,6 +70,9 @@ export class SkillCardImpl extends React.Component<SkillCardProps> {
                 <div className="skill-card-label">
                     <div className="skill-card-title">
                         {(status === "locked" || status === "completed") && <i className={`icon ${status === "locked" ? "lock" : "check circle"}`} />}
+                        {(status === "inprogress" || status === "notstarted") && maxSteps &&
```
Comment (shakao) [inline skillmap/src/components/SkillCard.tsx:73]: so "notstarted" shows 0/maxSteps?

File: skillmap/src/styles/skillcard.css [hunk 2]
```diff
@@ -182,6 +182,16 @@
     vertical-align: bottom;
 }
 
+.circular-label {
+	background: #656565;
```
Comment (shakao) [inline skillmap/src/styles/skillcard.css:186]: small: tabs instead of spaces?

**Review by shakao — APPROVED** [review]

File: skillmap/src/styles/skillcard.css (hunk 2 above)
Comment (riknoll) [reply skillmap/src/styles/skillcard.css:186]: HOW darn it.

File: skillmap/src/components/SkillCard.tsx (hunk 1 above)
Comment (riknoll) [reply skillmap/src/components/SkillCard.tsx:73]: oh no it doesn't but this is definitely confusing so I'll change it!

Comment (shakao) [reply skillmap/src/components/SkillCard.tsx:73]: ohh notstarted doesn't have maxsteps, i see

**Pull request merged**